platform/ATCmdParser.cpp

```diff
@@ -380,3 +380,44 @@ void ATCmdParser::abort()
 {
     _aborted = true;
 }
+
+bool ATCmdParser::process_oob()
```

Comment:
Do you have any test code for this ? I wish to see how are you using it.
It looks rather weird to me that you will have a Null at index 0 which means you have essentially set the pointer to _buffer as Null.

Reply:
@kjbracey-arm could you please review it ?

Reply:
This is a merge from ATParser which is already tested with ATParser, please see https://github.com/ARMmbed/ATParser/pull/14

Reply:
The code that you introduced is already a part of ATCmdParser in vrecv function.
https://github.com/ARMmbed/mbed-os/blob/master/platform/ATCmdParser.cpp#L274
Why would you do that again in a separate method ?